src/EnergyPlus/SystemReports.cc

```diff
@@ -3183,36 +3158,11 @@ void CreateEnergyReportStructure(EnergyPlusData &state)
             // Report selection
             ReportLoopData *select_ThisReportData(nullptr);
 
+            assert(LoopSideNum == LoopSideLocation::Demand || LoopSideNum == LoopSideLocation::Supply);
             if (PlantLoopNum <= state.dataHVACGlobal->NumPlantLoops) {
-                switch (LoopSideNum) {
-                case LoopSideLocation::Demand: {
-                    select_ThisReportData = &state.dataPlnt->VentRepPlantDemandSide(PlantLoopNum);
-                    break;
-                }
-                case LoopSideLocation::Supply: {
-                    select_ThisReportData = &state.dataPlnt->VentRepPlantSupplySide(PlantLoopNum);
-                    break;
-                }
-                default: {
-                    assert(false);
-                    break;
-                }
-                }
+                select_ThisReportData = &state.dataPlnt->VentRepPlant[static_cast<int>(LoopSideNum)](PlantLoopNum);
             } else { // CondLoop
-                switch (LoopSideNum) {
-                case LoopSideLocation::Demand: {
-                    select_ThisReportData = &state.dataPlnt->VentRepCondDemandSide(PlantLoopNum - state.dataHVACGlobal->NumPlantLoops);
-                    break;
-                }
-                case LoopSideLocation::Supply: {
-                    select_ThisReportData = &state.dataPlnt->VentRepCondSupplySide(PlantLoopNum - state.dataHVACGlobal->NumPlantLoops);
-                    break;
-                }
-                default: {
-                    assert(false);
-                    break;
-                }
-                }
+                select_ThisReportData = &state.dataPlnt->VentRepCond[static_cast<int>(LoopSideNum)](PlantLoopNum - state.dataHVACGlobal->NumPlantLoops);
             }
```

Comment:
Thank you, this was beautiful!

Reply:
You are welcome! Pretty much any time you can replace control flow (e.g., if or switch) with data indirection you want to do it. As with anything, you can go over the top with this, but EnergyPlus in general is pretty far from that point.

Reply:
Got it. I will keep it in mind!